### PR TITLE
Move Marshal::StringTo* methods to shared partition

### PR DESCRIPTION
**Pull request by marek-safar**

### CODE REVIEW

**Comment by marek-safar** [comment]
@jkotas I didn't change the overflow checks logic. Is it worth updating it to 64bits?

File: src/System.Private.CoreLib/shared/System/Runtime/InteropServices/Marshal.cs [hunk 1]
```diff
@@ -651,6 +651,134 @@ public static IntPtr SecureStringToGlobalAllocUnicode(SecureString s)
...
+            {
+                throw new ArgumentOutOfRangeException(nameof(s));
+            }
+
+            IntPtr hglobal = AllocHGlobal((IntPtr)nb);
+            
+            fixed (char* firstChar = s)
+            {
+                string.wstrcpy((char*)hglobal, firstChar, s.Length + 1);
+            }
```
Comment (stephentoub) [inline src/System.Private.CoreLib/shared/System/Runtime/InteropServices/Marshal.cs]: We could use span to make it a bit safer/shorter?
```C#
s.AsSpan().CopyTo(new Span<char>((char*)hglobal, nb));
```

Comment (stephentoub) [reply src/System.Private.CoreLib/shared/System/Runtime/InteropServices/Marshal.cs]: Hmm, nevermind, that won't copy the \0.  Probably simpler to stick with what you have.

**Comment by jkotas** [comment]
> Is it worth updating it to 64bits?

The interop refuses to work with large strings. The cut-off is different in different places. In general, if you have string in the GB range, your millage will vary. We really only want to make sure that these strings won't cause buffer overrun somewhere. It is fine to throw exceptions for these.

I think the most severe problem of these methods is up to 3x pessimistic over-allocation. It makes it useless for anything large anyway.

**Pull request merged**